sdk/identity/Azure.Identity/src/Validations.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Identity
+{
+    internal static class Validations
+    {
+
+
+
+        private const string InvalidTenantIdErrorMessage = "Invalid tenant id provided. You can locate your tenant id by following the instructions listed here: https://docs.microsoft.com/partner-center/find-ids-and-domain-names";
+
+        private const string NullTenantIdErrorMessage = "Tenant id cannot be null. You can locate your tenant id by following the instructions listed here: https://docs.microsoft.com/partner-center/find-ids-and-domain-names";
+
+        private const string NonTlsAuthorityHostErrorMessage = "Authority host must be a TLS protected (https) endpoint.";
+
+        /// <summary>
+        /// As tenant id is used in constructing authority endpoints and in command line invocation we validate the character set of the tenant id matches allowed characters.
+        /// </summary>
+        public static string ValidateTenantId(string tenantId, string argumentName = default, bool allowNull = false)
+        {
+            if (tenantId != null)
+            {
+                if (tenantId.Length == 0)
+                {
+                    throw (argumentName != null) ? new ArgumentException(InvalidTenantIdErrorMessage, argumentName) : new ArgumentException(InvalidTenantIdErrorMessage);
+                }
+
+                foreach (char c in tenantId)
+                {
+                    if (!IsValidTenantCharacter(c))
```

Comment:
logic is fine. 
using a regex to match might be better from perf perspective.

Reply:
There's not any performance gain in this case using regex, plus we would need to compile the regex and cache it statically. Since the character set is fairly simple in this case I chose to manually validate as it feels more explicit to me.